src/crimson/osd/osd.cc

```diff
@@ -472,11 +472,15 @@ seastar::future<bufferlist> OSD::load_map_bl(epoch_t e)
 
 seastar::future<std::unique_ptr<OSDMap>> OSD::load_map(epoch_t e)
 {
-  return load_map_bl(e).then([e, this](bufferlist bl) {
-    auto o = std::make_unique<OSDMap>();
-    o->decode(bl);
-    return seastar::make_ready_future<decltype(o)>(std::move(o));
-  });
+  auto o = std::make_unique<OSDMap>();
+  if (e > 0) {
+    return load_map_bl(e).then([e, o=std::move(o), this](bufferlist bl) mutable {
+      o->decode(bl);
+      return seastar::make_ready_future<unique_ptr<OSDMap>>(std::move(o));
+    });
+  } else {
+    return seastar::make_ready_future<unique_ptr<OSDMap>>(std::move(o));
```

Comment:
@liu-chunmei i pushed the change down to `OSD::load_map()`.

Reply:
@tchaikov e should be >=1. if e =1, not load map, see the origin logic .  I am not sure if your OSD::load_map only for  this  OSD::store_maps,   if  you want it for other cases, I am not sure if e= 1 should load map or not.

Reply:
and load_map_bl (e) should be load_map-bl(e-1).

Reply:
👍 fixed and repushed.

Reply:
That seems ok, I still debug on vstart start crimson-osd, after store_maps, program is  going to shutdown, don't know what except happened, tracing on it.